### PR TITLE
feat(ContentSecurityPolicy): Add a content security policy (#2094)

### PR DESCRIPTION
**Pull request by kotva006**
Re-Adding the CSP with more resources covered
https://app.asana.com/0/555089885850811/1207405407228256

### CODE REVIEW

File: config/runtime.exs [hunk 1]
```diff
@@ -229,12 +229,12 @@ case config_env() do
...
+               "img-src 'self' cdn.mbta.com #{System.get_env("STATIC_HOST", "")} #{System.get_env("CMS_API_BASE_URL", "")} *.google.com *.googleapis.com *.gstatic.com *.s3.amazonaws.com data: i.ytimg.com www.googletagmanager.com",
                "style-src 'self' 'unsafe-inline' www.gstatic.com #{System.get_env("STATIC_HOST", "")}",
-               "script-src 'self' 'unsafe-eval' 'unsafe-inline' #{System.get_env("STATIC_HOST", "")} translate.google.com www.gstatic.com www.googletagmanager.com *.googleapis.com",
+               "script-src 'self' 'unsafe-eval' 'unsafe-inline' #{System.get_env("STATIC_HOST", "")} www.instagram.com *.google.com www.gstatic.com www.googletagmanager.com *.googleapis.com data.mbta.com",
                "font-src 'self' #{System.get_env("STATIC_HOST", "")}",
                "connect-src 'self' *.googleapis.com #{sentry_dsn_host || ""}",
-               "frame-src 'self'"
+               "frame-src 'self' data.mbta.com data.mbta.com www.youtube.com www.google.com cdn.knightlab.com livestream.com www.instagram.com"
```
Comment (thecristen) [inline config/runtime.exs:237]: **typo**: got data.mbta.com on here twice

File: config/runtime.exs [hunk 2]
```diff
@@ -229,12 +229,12 @@ case config_env() do
            Enum.join(
              [
                "default-src 'none'",
-               "img-src 'self' cdn.mbta.com #{System.get_env("STATIC_HOST", "")} #{System.get_env("CMS_API_BASE_URL", "")} *.googleapis.com *.gstatic.com *.s3.amazonaws.com data:",
+               "img-src 'self' cdn.mbta.com #{System.get_env("STATIC_HOST", "")} #{System.get_env("CMS_API_BASE_URL", "")} *.google.com *.googleapis.com *.gstatic.com *.s3.amazonaws.com data: i.ytimg.com www.googletagmanager.com",
                "style-src 'self' 'unsafe-inline' www.gstatic.com #{System.get_env("STATIC_HOST", "")}",
-               "script-src 'self' 'unsafe-eval' 'unsafe-inline' #{System.get_env("STATIC_HOST", "")} translate.google.com www.gstatic.com www.googletagmanager.com *.googleapis.com",
+               "script-src 'self' 'unsafe-eval' 'unsafe-inline' #{System.get_env("STATIC_HOST", "")} www.instagram.com *.google.com www.gstatic.com www.googletagmanager.com *.googleapis.com data.mbta.com",
                "font-src 'self' #{System.get_env("STATIC_HOST", "")}",
                "connect-src 'self' *.googleapis.com #{sentry_dsn_host || ""}",
```
Comment (thecristen) [inline config/runtime.exs:236]: I believe `analytics.google.com` needs to be added here to `connect-src` too

File: config/runtime.exs [hunk 3]
```diff
@@ -229,12 +229,12 @@ case config_env() do
            Enum.join(
              [
                "default-src 'none'",
-               "img-src 'self' cdn.mbta.com #{System.get_env("STATIC_HOST", "")} #{System.get_env("CMS_API_BASE_URL", "")} *.googleapis.com *.gstatic.com *.s3.amazonaws.com data:",
+               "img-src 'self' cdn.mbta.com #{System.get_env("STATIC_HOST", "")} #{System.get_env("CMS_API_BASE_URL", "")} *.google.com *.googleapis.com *.gstatic.com *.s3.amazonaws.com data: i.ytimg.com www.googletagmanager.com",
```
Comment (thecristen) [inline config/runtime.exs:232]: question: do we need the `*.s3.amazonaws.com` one? As far as I could tell the only place we need this is for rendering map image tiles on dev (that URL is elsewhere in our config, `mbta-map-tiles-dev.s3.amazonaws.com`).

**Review by thecristen — CHANGES_REQUESTED** [review]

File: config/runtime.exs (hunk 2 above)
Comment (thecristen) [reply config/runtime.exs:236]: whoops looks like it's `www.google-analytics.com` (and needs to be in `script-src` as well as `connect-src`)

**Review by thecristen — CHANGES_REQUESTED** [review]
From the homepage

image

From the Sumner tunnel page (`/diversions/alternative-travel-options-during-the-2024-sumner-tunnel-closure`)

image

**Comment by kotva006** [comment]
Screenshot 2024-06-24 at 1 10 09 PM
Homepage

**Review by thecristen — APPROVED** [review]
let's give it a go! 🚀

**Pull request merged**